### PR TITLE
:ambulance: fixed developer mode availability for not admins via drop…

### PR DESCRIPTION
**Pull request by KolushovAlexandr**
…down menu

### CODE REVIEW

File: web_debranding/models/res_users.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+
+class ResUsers(models.Model):
+    _inherit = 'res.users'
+
+    @api.multi
+    def is_admin(self):
+        # By default Python functions starting with _ are considered private methods.
+        # Private methods (such as _is_admin) cannot be called remotely
+        return self._is_admin()
```
Comment (yelizariev) [inline web_debranding/models/res_users.py:14]: :+1:

File: web_debranding/static/src/js/user_menu.js [hunk 2]
```diff
@@ -9,6 +9,25 @@ odoo.define('web_debranding.UserMenu', function (require) {
 
     var UserMenu = require('web.UserMenu');
     UserMenu.include({
+        start: function () {
+            var self = this;
+            var result = this._super();
+
+            this._rpc({
```
Comment (yelizariev) [inline web_debranding/static/src/js/user_menu.js]: You must call it only after ``result.then(...)``

**Review by yelizariev — CHANGES_REQUESTED** [review]

**Review by yelizariev — CHANGES_REQUESTED** [review]
Add unit tests

File: web_debranding/tests/test_developer_mode_icon.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        code = """
+                    if ($('li a[data-menu="debug"]').length > 0 || $('li a[data-menu="debugassets"]').length > 0) {
+                        console.log('error', 'Developer mode icons are displayed for not admin user');
+                    } else {
+                        console.log('ok');
+                    }
+        """
+        self.phantom_js(url, code, login="demo")
+
+    def test_02_remove_add_to_cart(self):
```
Comment (yelizariev) [inline web_debranding/tests/test_developer_mode_icon.py]: cart?

**Review by yelizariev — CHANGES_REQUESTED** [review]

File: web_debranding/tests/test_developer_mode_menu_elements.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    def test_01_remove_developer_mode_menu_elements(self):
+
+        url = '/web'
+        code = """
+                    setTimeout(function(){
+                        if ($('li a[data-menu="debug"]').length > 0 || $('li a[data-menu="debugassets"]').length > 0) {
+                            console.log('error', 'Developer mode menu elements are displayed for not admin user');
+                        } else {
+                            console.log('ok');
```
Comment (yelizariev) [inline web_debranding/tests/test_developer_mode_menu_elements.py]: You must check that the page is load and checking for some elements makes sense

File: web_debranding/tests/test_developer_mode_menu_elements.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+        url = '/web'
+        code = """
+                    var k = 0;
+                    setInterval(function(){
+                        if ($('li a[data-menu="debug"]').length > 0 && $('li a[data-menu="debugassets"]').length > 0) {
+                            console.log('ok');
+                            return;
+                        } else if (k > 30){
+                            console.log('error', 'Developer mode menu elements are not displayed for admin');
```
Comment (yelizariev) [inline web_debranding/tests/test_developer_mode_menu_elements.py]: No need to wait 30*interval  ms each time. You only need to wait when page is loaded and then check condition once

**Review by yelizariev — CHANGES_REQUESTED** [review]

File: web_debranding/tests/test_developer_mode_menu_elements.py [hunk 6]
```diff
@@ -2,40 +2,48 @@
...
         url = '/web'
         code = """
-                    setTimeout(function(){
-                        if ($('li a[data-menu="debug"]').length > 0 || $('li a[data-menu="debugassets"]').length > 0) {
-                            console.log('error', 'Developer mode menu elements are displayed for not admin user');
-                        } else {
-                            console.log('ok');
-                        }
-                    }, 10000)
+                    $(document).ready( function() {
```
Comment (yelizariev) [inline web_debranding/tests/test_developer_mode_menu_elements.py]: `` $(document).ready`` may not be enough if target html rendered via js and not on server side

File: web_debranding/tests/test_developer_mode_menu_elements.py [hunk 7]
```diff
@@ -2,40 +2,48 @@
...
+                        setTimeout(function(){
+                            if ($('li a[data-menu="debug"]').length > 0 || $('li a[data-menu="debugassets"]').length > 0) {
+                                console.log('error', 'Developer mode menu elements are displayed for not admin user');
+                            } else {
+                                console.log('ok');
+                            }
+                        }, 3000);
```
Comment (yelizariev) [inline web_debranding/tests/test_developer_mode_menu_elements.py]: It's not stable solution

**Review by yelizariev — CHANGES_REQUESTED** [review]
Use setInterval and check that page is loaded (e.g. top menu is presented)

File: web_debranding/static/src/js/user_menu.js [hunk 8]
```diff
@@ -9,6 +13,28 @@ odoo.define('web_debranding.UserMenu', function (require) {
...
+                    method: "is_admin",
+                    args: [odoo.session_info.uid],
+                }).then(function(r){
+                    if (!r) {
+                        self.remove_debug_links();
+                    }
+                });
+            });
+
+            return result;
```
Comment (yelizariev) [inline web_debranding/static/src/js/user_menu.js]: I am not sure how it works, ``.then`` may return a new object and doesn't modify ``result``. If so, you return wrong object

**Review by yelizariev — CHANGES_REQUESTED** [review]
Tests looks fine

File: web_debranding/tests/__init__.py [hunk 9]
```diff
@@ -0,0 +1,3 @@
+# License LGPL - 3.0 or later(https: // www.gnu.org / licenses / lgpl.html).
```
Comment (yelizariev) [inline web_debranding/tests/__init__.py]: Why so strange link?

**Review by yelizariev — CHANGES_REQUESTED** [review]

File: web_debranding/__init__.py [hunk 10]
```diff
@@ -1,3 +1,5 @@
+# License LGPL - 3.0 or later(https: // www.gnu.org / licenses / lgpl.html).
```
Comment (yelizariev) [inline web_debranding/__init__.py]: Fix licenses

File: web_debranding/models/__init__.py [hunk 11]
```diff
@@ -1,7 +1,10 @@
+# License LGPL - 3.0 or later(https: // www.gnu.org / licenses / lgpl.html).
```
Comment (yelizariev) [inline web_debranding/models/__init__.py]: Fix licenses

Comment (yelizariev) [reply web_debranding/models/__init__.py]: Fix other spaces too. Compare your comment with 

    # License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).

**Review by yelizariev — CHANGES_REQUESTED** [review]

**Review by yelizariev — APPROVED** [review]

**Review by ilmir-k — APPROVED** [review]

**Pull request merged**